bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/Hors.java

```diff
@@ -0,0 +1,171 @@
+        urlopen = new Urllib(context);
+        urlopen.setAllowCircularRedirects(true);
+        response = urlopen.open("https://www.dittkort.se//q/?p=7EB4F129-0A41-417F-8FEA-51B2B75B9D24");
+
+        Document d = Jsoup.parse(response);
+        Element e = d.getElementById("__VIEWSTATE");
+        if (e == null || e.attr("value") == null) {
+            throw new BankException(
+                    res.getText(R.string.unable_to_find).toString() + " ViewState.");
+        }
+        String viewState = e.attr("value");
+
+        e = d.getElementById("__EVENTVALIDATION");
+        if (e == null || e.attr("value") == null) {
+            throw new BankException(
+                    res.getText(R.string.unable_to_find).toString() + " EventValidation.");
+        }
+        String eventValidation = e.attr("value");
+
+        List<NameValuePair> postData = new ArrayList<NameValuePair>();
+        postData.add(new BasicNameValuePair("__EVENTTARGET", ""));
+        postData.add(new BasicNameValuePair("__EVENTARGUMENT", ""));
+        postData.add(new BasicNameValuePair("__VIEWSTATE", viewState));
+        postData.add(new BasicNameValuePair("__VIEWSTATEGENERATOR", "D7823E0B"));
+        postData.add(new BasicNameValuePair("__EVENTVALIDATION", eventValidation));
+        postData.add(new BasicNameValuePair("_ctl0:cphMain:SmartpassTextbox", getUsername()));
+        postData.add(new BasicNameValuePair("_ctl0:cphMain:SubmitButton", "OK"));
+        postData.add(new BasicNameValuePair("_ctl0:cphMain:chbRememberSmartPassCode", "on"));
+        postData.add(new BasicNameValuePair("_ctl0:cphMain:cookieEnabled", "true"));
+        return new LoginPackage(urlopen, postData, response, "https://www.dittkort.se//q/?p=7EB4F129-0A41-417F-8FEA-51B2B75B9D24");
+    }
+
+    public Urllib login() throws LoginException, BankException, IOException {
+        LoginPackage lp = preLogin();
+        response = urlopen.open(lp.getLoginTarget(), lp.getPostData());
+        if (response.contains("cphMain_txtGeneralCardError")) {
+            throw new LoginException(res.getText(R.string.invalid_card_number).toString());
+        }
+        return urlopen;
+    }
+
+    @Override
+    public void update() throws BankException, LoginException, BankChoiceException, IOException {
+        super.update();
+        if (getUsername().isEmpty()) {
+            throw new LoginException(res.getText(R.string.invalid_card_number).toString());
+        }
+        urlopen = login();
+        Document document = Jsoup.parse(response);
+        Element balanceElement = document.getElementById("cphMain_lblAmount");
+        if (balanceElement == null) {
+            throw new BankException(
+                    res.getText(R.string.unable_to_find).toString() + res.getText(R.string.balance).toString());
+        }
+
+        Element nameElement = document.getElementById("lblCardName");
+        String accountName = NAME.toUpperCase();
```

Comment:
Could be replaced by `String accountName = nameElement == null ? NAME.toUpperCase() : nameElement.text()`